src/rust/ide/src/ide/integration.rs

```diff
@@ -286,6 +287,22 @@ impl Integration {
             eval_ project_frp.editing_aborted   (invalidate.trigger.emit(()));
             eval_ project_frp.save_module       (model.module_saved_in_ui());
         }
+
+        frp::extend! {network
+            eval editor_outs.node_added([](_){analytics::remote_log(analytics::data::Public::new("node_added"))});
```

Comment:
few things:
1. Use `eval_`
2. expose `Public` inside `analytics`, so we can write `analytics::Public` instead.
3. Add space before "network"
4. Some events start with "on", some do not, this is very strange. Remove "on", because we log only events there.
5. Why do you think that logging it fro "integration" is better idea than logging it directly from components like "graph_editor"? I'm still unsure what to think about it.

Reply:
Re 5) I have no string opinion. I had expected that error logging also needs to be in the controller, but that moved even higher to the JS side. This snippet could just as well be inside the graph editor.